fortnite_api/stats.py

```diff
@@ -1,65 +1,261 @@
+    trio: Optional[:class:`BrGameModeStats`]
+        The player's stats for trio game modes. This is ``None`` if the player has no stats for trio game modes.
+    squad: Optional[:class:`BrGameMode`]
+        The player's stats for squad game modes. This is ``None`` if the player has no stats for squad game modes.
+    """
+
+    __slots__: Tuple[str, ...] = ('overall', 'solo', 'duo', 'trio', 'squad', 'raw_data')
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
+        _overall = data.get('overall')
+        self.overall: Optional[BrGameModeStats] = _overall and BrGameModeStats(data=_overall)
 
-    def __init__(self, data):
-        self.overall = BrGameModeStats(data.get('overall')) if data.get('overall') else None
-        self.solo = BrGameModeStats(data.get('solo')) if data.get('solo') else None
-        self.duo = BrGameModeStats(data.get('duo')) if data.get('duo') else None
-        self.trio = BrGameModeStats(data.get('trio')) if data.get('trio') else None
-        self.squad = BrGameModeStats(data.get('squad')) if data.get('squad') else None
-        self.raw_data = data
+        _solo = data.get('solo')
+        self.solo: Optional[BrGameModeStats] = _solo and BrGameModeStats(data=_solo)
+
+        _duo = data.get('duo')
+        self.duo: Optional[BrGameModeStats] = _duo and BrGameModeStats(data=_duo)
+
+        _trio = data.get('trio')
+        self.trio: Optional[BrGameModeStats] = _trio and BrGameModeStats(data=_trio)
+
+        _squad = data.get('squad')
+        self.squad: Optional[BrGameModeStats] = _squad and BrGameModeStats(data=_squad)
 
 
 class BrGameModeStats:
+    """Represents the specific stats for a Fortnite player in a specific game mode on a specific input type.
+
+    Attributes
+    ----------
+    score: :class:`int`
+        The total score for the stats for a specific game mode.
+    score_per_min: :class:`float`
+        The score per minute for the stats for a specific game mode. This is the score divided by the minutes played.
+    score_per_match: :class:`float`
+        The score per match for the stats for a specific game mode. This is the score divided by the matches played.
+    wins: :class:`int`
+        The total number of wins in this specific game mode.
+    top3: :class:`int`
+        The number of times the player has placed in the top 3 in this specific game mode.
+    top5: :class:`int`
+        The number of times the player has placed in the top 5 in this specific game mode.
+    top6: :class:`int`
+        The number of times the player has placed in the top 6 in this specific game mode.
+    top10: :class:`int`
+        The number of times the player has placed in the top 10 in this specific game mode.
+    top12: :class:`int`
+        The number of times the player has placed in the top 12 in this specific game mode.
+    top25: :class:`int`
+        The number of times the player has placed in the top 25 in this specific game mode.
```

Comment:
We can be more specific in the docs. Specific fields are only set for specific team sizes:
### 10 players remaining:
- top10 = Solo
- top5 = Duo
- top3 = Squad
### 25 players remaining:
- top25 = Solo
- top12 = Duo
- top6 = Squad

Reply:
Updated in [801d6ed](https://github.com/Fortnite-API/py-wrapper/pull/19/commits/801d6edb6c26c2416798561752b2a1137ea6672f) - this appears to have happened because of a major oversight due to no stat tests. Adding stat tests now.